adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/HttpClientWrapper.cs

```diff
@@ -0,0 +1,192 @@
+                                this.ContentType);
+                        }
+                        else
+                        {
+                            content = new FormUrlEncodedContent(((DictionaryRequestParameters) this.BodyParameters)
+                                .ToList());
+                        }
+
+                        requestMessage.Method = HttpMethod.Post;
+                        requestMessage.Content = content;
+                    }
+                    else
+                    {
+                        requestMessage.Method = HttpMethod.Get;
+                    }
+
+                    responseMessage = await client.SendAsync(requestMessage).ConfigureAwait(false);
+                }
+                catch (TaskCanceledException ex)
+                {
+                    throw new HttpRequestWrapperException(null, ex);
+                }
+
+                IHttpWebResponse webResponse = await CreateResponseAsync(responseMessage).ConfigureAwait(false);
+
+                if (!responseMessage.IsSuccessStatusCode)
+                {
+                    throw new HttpRequestWrapperException(webResponse, new HttpRequestException(
+                        string.Format(CultureInfo.CurrentCulture,
+                            "Response status code does not indicate success: {0} ({1}).",
+                            (int) webResponse.StatusCode, webResponse.StatusCode),
+                        new AdalException(webResponse.ResponseString)));
+                }
+
+                if (addCorrelationId)
+                {
+                    VerifyCorrelationIdHeaderInReponse(webResponse.Headers);
+                }
+
+                return webResponse;
+            }
+        }
+
+        public async static Task<IHttpWebResponse> CreateResponseAsync(HttpResponseMessage response)
+        {
+            return new HttpWebResponseWrapper(await response.Content.ReadAsStringAsync().ConfigureAwait(false), response.Headers,
+                response.StatusCode);
+        }
+
+        private void VerifyCorrelationIdHeaderInReponse(HttpResponseHeaders headers)
+        {
+            foreach (KeyValuePair<string, IEnumerable<string>> header in headers)
+            {
+                string reponseHeaderKey = header.Key;
+                string trimmedKey = reponseHeaderKey.Trim();
+                if (string.Compare(trimmedKey, OAuthHeader.CorrelationId, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    string correlationIdHeader = headers.GetValues(trimmedKey).FirstOrDefault().Trim();
+                    Guid correlationIdInResponse;
+                    if (!Guid.TryParse(correlationIdHeader, out correlationIdInResponse))
+                    {
+                        var msg = string.Format(CultureInfo.CurrentCulture,
+                            "Returned correlation id '{0}' is not in GUID format.", correlationIdHeader);
```

Comment:
>"Returned correlation id '{0}' is not in GUID format." [](start = 28, length = 54)

any reason why strings are not added to resources?

Reply:
I add reusable strings to a resource/constants file. If they are only used once then, personally, I feel there is no specific advantage to moving that string to a constants file.

---
In reply to: [160296794](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/937#discussion_r160296794) [](ancestors = 160296794)